### ISSUE TITLE
torch.include and local module variables

### ISSUE THREAD

**Issue by nicholas-leonard** [torch/torch7#525]
I can't seem to find the doc for this function.

On a related note, I know it is good practice to not make new package modules global. So the idea would be to have require return module as a local variable : 
```lua
local mypkg = require 'mypkg'
```

Such that that package's init.lua file would look like this : 
```lua
local mypkg = {}

-- add stuff to mypkg

return mypkg
```

However, I have yet to get this working. I know about hacking torch.class to use local modules : 
```lua
local myclass, parent = torch.class("mypkg.MyClass", "mypkg.Parrent", mypkg)
```

So if `myclass` is stored in `myclass.lua`, I need to be able to load it from the init.lua (in the dostuff section) . With globals, I do : 

```lua
torch.include('mypkg', myclass.lua)
```

But calling the above with a `local mypkg` instead of a global, I get this kind of error : 
```bash
/usr/local/share/lua/5.1/mypkg/config.lua:4: attempt to index global 'mypkg' (a nil value)
stack traceback:
	/usr/local/share/lua/5.1/mypkg/config.lua:4: in main chunk
	[C]: in function 'dofile'
	/usr/local/share/lua/5.1/torch/init.lua:54: in function 'include'
	/usr/local/share/lua/5.1/mypkg/init.lua:15: in main chunk
	[C]: in function 'require'
	stdin:1: in main chunk
	[C]: at 0x00405e60
```

So basically, because dl is local to init.lua, it doesn't exist in `myclass.lua`.

I should mention that I am trying to get this working after a `luarocks make rocks/mypkg-scm-1.rockspec`.

So is there a way around this or I am going to have to continue to make my packages global variables?

**Comment by soumith** [comment]
use this trick: 
https://github.com/facebook/iTorch/blob/master/_env.lua
(just make it local here) https://github.com/facebook/iTorch/blob/master/init.lua#L10
https://github.com/facebook/iTorch/blob/master/bokeh.lua#L15

**Comment by nicholas-leonard** [comment]
@soumith Smartest thing I have heard all day :+1:
